package/gluon-status-page-api/src/stations.c

```diff
@@ -40,12 +45,75 @@ static void badrequest() {
   exit(1);
 }
 
+// recurse down to the lowest layer-2 interface
+static int interface_get_lowest(const char *ifname, char *hwifname);
+static int interface_get_lowest(const char *ifname, char *hwifname) {
+  glob_t globbuf;
+  char *fnamebuf = alloca(1 + strlen(VIRTIF_PREFIX) + IF_NAMESIZE +
+                          strlen(LOWERGLOB_SUFFIX));
+  char *lowentry = NULL;
+
+
+  sprintf(fnamebuf, "%s%s%s", VIRTIF_PREFIX, ifname, LOWERGLOB_SUFFIX);
+  glob(fnamebuf, GLOB_NOSORT, NULL, &globbuf);
+
+  if (globbuf.gl_pathc == 1) {
+    lowentry = alloca(1 + strlen(globbuf.gl_pathv[0]));
+    strncpy(lowentry, globbuf.gl_pathv[0], 1 + strlen(globbuf.gl_pathv[0]));
+  }
+
+  globfree(&globbuf);
+
+  if (!lowentry) {
+    char *path = alloca(1 + strlen(NETIF_PREFIX) + strlen(ifname));
+    sprintf(path, "%s%s", NETIF_PREFIX, ifname);
+
+    if(access(path, F_OK) != 0)
+      return false;
+
+    strncpy(hwifname, ifname, IF_NAMESIZE - 1);
+    return true;
+  } else {
+    char buf[PATH_MAX];
+    ssize_t len;
+
+    if ((len = readlink(lowentry, buf, sizeof(buf)-1)) != -1)
+      buf[len] = '\0';
+    else
+      return false;
+
+    if (buf[0] == '.' && buf[1] == '.' && buf[2] == '/') {
+      return interface_get_lowest(strrchr(buf, '/') + 1, hwifname);
+    } else {
+      return false;
+    }
+  }
+}
+
 int main(int argc, char *argv[]) {
   if (argc != 2)
     badrequest();
 
   const char *ifname = argv[1];
-  const struct iwinfo_ops *iw = iwinfo_backend(ifname);
+  char hwifname[IF_NAMESIZE] = "";
+
+  if (strlen(ifname) > IF_NAMESIZE)
+    badrequest();
+
+  if (strchr(ifname, '/') != NULL ||
+      strchr(ifname, '\\') != NULL ||
+      strchr(ifname, '[') != NULL ||
+      strchr(ifname, ']') != NULL ||
+      strchr(ifname, '{') != NULL ||
+      strchr(ifname, '}') != NULL ||
+      strchr(ifname, '*') != NULL ||
+      strchr(ifname, '?') != NULL)
```

Comment:
As far as I understand `strspn(3)`, it can't, because `strspn()` only looks at the beginning of `ifname` while `strchr()` looks anywhere in the string.